UnleashClient/features/Feature.py

```diff
@@ -110,19 +105,53 @@ def get_variant(self, context: dict = None) -> dict:
         :param context: Context information
         :return:
         """
-        variant = DISABLED_VARIATION
-        is_feature_enabled = self.is_enabled(context)
-
-        if is_feature_enabled and self.variants is not None:
+        evaluation_result = self._get_evaluation_result(context)
+        variant = None
+        is_feature_enabled = False
+
+        if evaluation_result is not None:
+            variant = evaluation_result.variant
+            is_feature_enabled = evaluation_result.enabled
+
+        if (
+            is_feature_enabled
+            and self.variants is not None
+            and (variant is None or variant == DISABLED_VARIATION)
```

Comment:
Do we get back the disabled variant or a non variant here?  From a look over the code, the None check here shouldn't ever happen